### PR TITLE
Set re-prompts and hints into session during confused bot state

### PR DESCRIPTION
**Pull request by LaurenceTews**

### CODE REVIEW

File: mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java [hunk 1]
```diff
@@ -141,7 +142,7 @@ public InkBot(T configuration)
     List<String> defaultResponses = configuration.getDefaultResponses();
     if (defaultResponses != null)
     {
-      setDefaultResponses(defaultResponses.toArray(new String[defaultResponses.size()]));
+      setDefaultResponses(defaultResponses.toArray(new String[0]));
```
Comment (rabidgremlin) [inline mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java]: Why did you change this ? I believe it causes a double allocation of the array ?

File: mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java [hunk 2]
```diff
@@ -156,7 +157,7 @@ public BotResponse respond(Session session, Context context, String messageText)
     throws BotException
   {
     log.debug("===> \n session: {} context: {} messageText: {}",
-        new Object[]{ session, context, messageText });
+            session, context, messageText);
```
Comment (rabidgremlin) [inline mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java]: Need the Object[] bit as SLF4J otherwise doesn't log correctly.... (unless they have changed this is later versions ?)

File: mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java (hunk 1 above)
Comment (LaurenceTews) [reply mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java]: As of Java8 this way of allocating arrays is the most efficient.

File: mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java (hunk 2 above)
Comment (LaurenceTews) [reply mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/InkBot.java]: have reverted.
